packages/monaco-editor-wrapper/src/monacoEditorWrapper.ts

```diff
@@ -0,0 +1,59 @@
+import { editor, languages } from 'monaco-editor/esm/vs/editor/editor.api.js';
+import { EditorConfig, MonacoEditorWrapper } from './wrapper.js';
+
+/**
+ * This is derived from:
+ * https://microsoft.github.io/monaco-editor/api/interfaces/monaco.languages.ILanguageExtensionPoint.html
+ */
+export type MonacoLanguageExtensionConfig = {
+    id: string;
+    extensions?: string[];
+    filenames?: string[];
+    filenamePatterns?: string[];
+    firstLine?: string;
+    aliases?: string[];
+    mimetypes?: string[];
+}
+
+export type DirectMonacoEditorWrapperConfig = {
+    languageExtensionConfig?: MonacoLanguageExtensionConfig;
+    languageDef?: languages.IMonarchLanguage;
+    themeData?: editor.IStandaloneThemeData;
+}
+
+export class DirectMonacoEditorWrapper implements MonacoEditorWrapper {
+
+    async init(editorConfig: EditorConfig, wrapperConfig: DirectMonacoEditorWrapperConfig) {
+        // register own language first
+        const extLang = wrapperConfig?.languageExtensionConfig;
+        if (extLang) {
+            languages.register(extLang);
+        }
+
+        const languageRegistered = languages.getLanguages().filter(x => x.id === editorConfig.languageId);
+        if (languageRegistered.length === 0) {
+            // this is only meaningful for languages supported by monaco out of the box
+            languages.register({ id: editorConfig.languageId });
+        }
+
+        // apply monarch definitions
+        const tokenProvider = wrapperConfig?.languageDef;
+        if (tokenProvider) {
+            languages.setMonarchTokensProvider(editorConfig.languageId, tokenProvider);
+        }
+        const themeData = wrapperConfig?.themeData;
+        if (themeData) {
+            editor.defineTheme(editorConfig.theme, themeData);
+        }
+        editor.setTheme(editorConfig.theme);
+
+        console.log('Init of MonacoConfig was completed.');
+        return Promise.resolve();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    async updateConfig(_options: editor.IEditorOptions & editor.IGlobalEditorOptions) {
+        // nothing
```

Comment:
Nothing? Can we not implement this method, or is it required for the class?

Reply:
This function does not make sense when using the classic monaco-editor configuration approach. I will change it
```typescript
class DirectMonacoEditorWrapper implements Partial<MonacoEditorWrapper>
```
The reconfiguration requires the real editor instance which they don't have. editor/editorOptions/modelRef belong togehter. Maybe it really makes sense to move them into one common base class. I will think about that.

Reply:
I have introduced an editor base class and move everything related to the editor from the wrapper there. The other classes extends this now. I also used the chance to give better names.